core/src/main/java/org/apache/iceberg/io/RollingDataWriter.java

```diff
@@ -41,7 +41,6 @@ public RollingDataWriter(FileWriterFactory<T> writerFactory, OutputFileFactory f
     super(fileFactory, io, targetFileSizeInBytes, spec, partition);
     this.writerFactory = writerFactory;
     this.dataFiles = Lists.newArrayList();
-    openCurrentWriter();
```

Comment:
`openCurrentWriter` calls `newWriter`, which is implemented by subclasses. In this case, it uses `writerFactory`. Calling `openCurrentWriter` in the parent class will cause `newWriter` to be called before setting `writerFactory`.

Reply:
Sorry, I ignored it. Thanks for pointing out my mistake.